x-pack/plugin/esql/build.gradle

```diff
@@ -99,11 +99,18 @@ tasks.named("test").configure {
     def imagesDocFolder = file("${rootDir}/docs/reference/query-languages/esql/images")
     def snippetsDocFolder = file("${rootDir}/docs/reference/query-languages/esql/_snippets")
     def kibanaDocFolder = file("${rootDir}/docs/reference/query-languages/esql/kibana")
+    def snippetsTree = fileTree(snippetsFolder).matching {
```

Comment:
The fix for --configuration-cache is essentially to move these calls to `fileTree` out of the `doLast` while still making sure they run after the tests are finished.